### PR TITLE
fix: e2e in metrics collection with runonserver query created by user

### PR DESCRIPTION
**Pull request by wadlejitendra**
This PR has been fixed following issues in metrics collection tests:

- `semantic version` range to limit the versions of PostgreSQL to run query on (`>=10.0.0`) in custom query configMap file.
-   regex keyword value length fix from `8 to 9` in metrics E2E. 

Closes: #366
Closes: #365 

Signed-off-by: Jitendra Wadle <jitendra.wadle@enterprisedb.com>

### CODE REVIEW

**Review by danishedb — APPROVED** [review]

**Review by armru — APPROVED** [review]

File: tests/e2e/metrics_test.go [hunk 1]
```diff
@@ -56,18 +57,20 @@ var _ = Describe("Metrics", func() {
...
+		"cnpg_pg_database_size_bytes{datname=\"app\"} [0-9e\\+\\.]+|",
+		"cnpg_pg_replication_slots_inactive 0|",
+		"cnpg_pg_stat_archiver_archived_count \\d+|",
+		"cnpg_pg_stat_archiver_failed_count \\d+|",
+		"cnpg_pg_locks_blocked_queries 0|",
+		"cnpg_runonserver_match_fixed 42|",
+		"cnpg_collector_last_collection_error 0)",
+	}
+
+	metricsRegexp := regexp.MustCompile(fmt.Sprintf(`(?m:^(`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`$)`,
```
Comment (leonardoce) [inline tests/e2e/metrics_test.go:72]: Unfortunately this regex is poorly readable and dependent on the length of `metricsList`.
I'd like, if possible, to have something that's more generic.
Perhaps splitting the metrics output on lines and counting just the metrics we are interested in would be better?

Comment (armru) [inline tests/e2e/metrics_test.go:72]: We should improve this

**Review by leonardoce — CHANGES_REQUESTED** [review]

File: tests/e2e/metrics_test.go (hunk 1 above)
Comment (wadlejitendra) [reply tests/e2e/metrics_test.go]: @leonardoce and @armru Fixed. 
Please review.

**Review by armru — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Pull request merged**